File: tests/fixtures/scala/UserController.scala
package com.example.app

import play.api.mvc._

/** A user controller. */
class UserController(val service: UserService) extends Controller {
  def show(id: Long) = Action { request =>
    Ok(service.find(id).toString)
  }

  def list = Action { Ok("list") }
}

trait UserService {
  def find(id: Long): User
}

object UserService {
  def apply(): UserService = new DefaultService()
}

case class User(id: Long, name: String)

object Helpers {
  def format(text: String): String = text.toUpperCase
}

File: tests/fixtures/kotlin/UserController.kt
package com.example.app

import org.springframework.web.bind.annotation.*

/**
 * REST controller for users.
 */
@RestController
@RequestMapping("/users")
public class UserController(private val service: UserService) {

    @GetMapping("/{id}")
    suspend fun getUser(id: Long): User {
        return service.find(id)
    }

    @PostMapping
    fun createUser(user: User): User {
        return user
    }
}

interface UserService {
    fun find(id: Long): User
}

data class User(val id: Long, val name: String)

object Helpers {
    fun format(text: String): String = text.uppercase()
}

@Composable
fun MyView(name: String) {
    Text(text = name)
}
